2005-06-11  GOTO Masanori  <gotom@debian.org>

	Enable ___arch__swab16, ___arch__swab32.
	Remove ___arch__swab64 because ppc64 does not have appropriate
	instruction for this purpose.
	Move byteorder-const.h ppc64 part to this file.

--- ../linux-2.6.12-rc6/include/asm-ppc64/byteorder.h	2005-06-07 00:22:29.000000000 +0900
+++ include/asm-ppc64/byteorder.h	2005-06-13 00:57:52.150659216 +0900
@@ -40,8 +40,7 @@
 	__asm__ __volatile__ ("stwbrx %1,0,%2" : "=m" (*addr) : "r" (val), "r" (addr));
 }
 
-#if 0
-static __inline__ __attribute_const__ __u16 ___arch__swab16(__u16 value)
+static __inline__ __u16 ___arch__swab16(__u16 value)
 {
 	__u16 result;
 
@@ -51,7 +50,7 @@
 	return result;
 }
 
-static __inline__ __attribute_const__ __u32 ___arch__swab32(__u32 value)
+static __inline__ __u32 ___arch__swab32(__u32 value)
 {
 	__u32 result;
 
@@ -63,17 +62,8 @@
 	return result;
 }
 
-static __inline__ __attribute_const__ __u64 ___arch__swab64(__u64 value)
-{
-	__u64 result;
-#error implement me
-}
-
 #define __arch__swab16(x) ___arch__swab16(x)
 #define __arch__swab32(x) ___arch__swab32(x)
-#define __arch__swab64(x) ___arch__swab64(x)
-
-#endif
 
 /* The same, but returns converted value from the location pointer by addr. */
 #define __arch__swab16p(addr) ld_le16(addr)
